pkg/plugins/volume/ceph/size.go

```diff
@@ -0,0 +1,125 @@
+// SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and IronCore contributors
+// SPDX-License-Identifier: Apache-2.0
+
+package ceph
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/ceph/go-ceph/rados"
+	"github.com/ceph/go-ceph/rbd"
+	"github.com/ironcore-dev/libvirt-provider/pkg/api"
+)
+
+func createKeyFile(imageName string, key string) (string, func() error, error) {
+	noOpCleanup := func() error { return nil }
+	file, err := os.CreateTemp("", fmt.Sprintf("%s-*", imageName))
+	if err != nil {
+		return "", noOpCleanup, fmt.Errorf("failed to create temp file: %w", err)
```

Comment:
return "", nil, fmt.Errorf(...) this cannot work?

Reply:
no, if you call a `nil` function you will see an exception.

Reply:
but you don't call it. if you return err, you don't continue in execution. Do you want to keep is more general for use?